rs/execution_environment/src/canister_manager.rs

```diff
@@ -2352,6 +2355,72 @@ impl CanisterManager {
             ),
         }
     }
+
+    /// Renames the canister from `old_id` to `new_id` and adds the appropriate entry into the canister history.
+    pub(crate) fn rename_canister(
+        &self,
+        sender: PrincipalId,
+        canister: &mut CanisterState,
+        origin: CanisterChangeOrigin,
+        old_id: CanisterId,
+        new_id: CanisterId,
+        to_version: u64,
+        to_total_num_changes: u64,
+        state: &mut ReplicatedState,
+    ) -> Result<(), CanisterManagerError> {
+        // In addition to this endpoint only being available from the NNS subnet, the calling canister
```

Comment:
Do we want to tighten the check even further to only allow a single canister ID on the NNS subnet to be the caller?